src/Sylius/Bundle/CoreBundle/CatalogPromotion/CommandHandler/TerminateCatalogPromotionHandler.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler;
+
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\RemoveCatalogPromotion;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\TerminateCatalogPromotion;
+use Sylius\Component\Core\Model\CatalogPromotion;
+use Sylius\Component\Promotion\Event\CatalogPromotionUpdated;
+use Sylius\Component\Promotion\Model\CatalogPromotionStates;
+use Sylius\Component\Promotion\Repository\CatalogPromotionRepositoryInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Stamp\DispatchAfterCurrentBusStamp;
+
+final class TerminateCatalogPromotionHandler
+{
+    public function __construct(
+        private CatalogPromotionRepositoryInterface $catalogPromotionRepository,
+        private MessageBusInterface $commandBus,
+        private MessageBusInterface $eventBus,
+    ) {
+    }
+
+    public function __invoke(TerminateCatalogPromotion $command): void
+    {
+        /** @var CatalogPromotion|null $catalogPromotion */
+        $catalogPromotion = $this->catalogPromotionRepository->findOneBy(['code' => $command->code]);
+
+        if (null === $catalogPromotion) {
+            return;
+        }
```

Comment:
IMO we should throw an exception here, the same in the `RemoveCatalogPromotionHandler`